File: Python/app/routes/product.py
from app.connection_db import supabase
from fastapi import APIRouter, Depends
from pydantic import BaseModel
from typing import Optional

class Product(BaseModel):
    id_produto: Optional[int] = None
    nome: str
    descricao: str
    preco: float
    estoque: int
    lote: int

class ProductCreate(BaseModel):
    nome: str
    descricao: str
    preco: float
    estoque: int
    lote: int

router = APIRouter()

@router.post('/product/post')
def insert_product(product: ProductCreate):
    data = supabase.table("produtos").insert(
        product.model_dump(exclude_none=True)
    ).execute()
    return {"Produto inserido": data}

@router.get('/product/get')
def get_product():
    data = supabase.table("produtos").select("*").execute()
    return {"produtos": data}

@router.put('/product/put/{id_produto}')
def update_product(id: int, update: ProductCreate):
    data = supabase.table("produtos").update(
        update.model_dump(exclude_none=True)
        ).eq("id_produto", id).execute()
    return {"Produto atualizado": data}    

@router.delete('/product/delete/{id_produto}')
def delete_product(id: int):
    data = supabase.table("produtos").delete().eq("id_produto", id).execute()
    return {"Produto deletado": data}

File: Flutter/lib/models/product.dart

class Product {
  final int idProduto;
  final String nome;
  final String descricao;
  final double preco;
  final int estoque;
  final int lote;
  final int categoriaProduto;

  Product({
    required this.idProduto,
    required this.nome,
    required this.descricao,
    required this.preco,
    required this.estoque,
    required this.lote,
    required this.categoriaProduto,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      idProduto: json['id_produto'] as int,
      nome: json['nome'] as String,
      descricao: json['descricao'] as String,
      preco: (json['preco'] as num).toDouble(),
      estoque: json['estoque'] as int,
      lote: json['lote'] as int,
      categoriaProduto: json['categoria_produto'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_produto': idProduto,
      'nome': nome,
      'descricao': descricao,
      'preco': preco,
      'estoque': estoque,
      'lote': lote,
      'categoria_produto': categoriaProduto,
    };
  }
}
